src/frontend/apps/impress/src/features/docs/doc-header/components/DocToolBox.tsx

```diff
@@ -31,6 +36,32 @@ export const DocToolBox = ({ doc, versionId }: DocToolBoxProps) => {
   const [isModalVersionOpen, setIsModalVersionOpen] = useState(false);
   const { isSmallMobile } = useResponsiveStore();
   const { authenticated } = useAuthStore();
+  const { docsStore } = useDocStore();
+  const { toast } = useToastProvider();
+
+  const copyCurrentEditorToClipboard = async (
+    asFormat: 'html' | 'markdown',
+  ) => {
+    const editor = docsStore[doc.id]?.editor;
+    if (!editor) {
+      toast(t('Editor unavailable'), VariantType.ERROR, { duration: 3000 });
+      return;
+    }
+
+    try {
+      const editorContentFormatted = await getEditorContentFormatted(
+        editor,
+        asFormat,
+      );
```

Comment:
I find this function (`getEditorContentFormatted`) a bit overkill, let's just have our condition here I think.